integrationTests/vm/arwen/testdata/erc20-c-03/elrond/util.h

```diff
@@ -0,0 +1,30 @@
+#include "context.h"
+#include "bigInt.h"
+
+
+// Macros to simplify error declaration and handling
+#define ERROR_MSG(var, str) \
+    const int var##_LEN = sizeof str - 1;\
+    byte var[var##_LEN + 1] = str;
+
+#define SIGNAL_ERROR(var) signalError(var, var##_LEN);
+
+ERROR_MSG(ERR_NUM_ARGS, "wrong number of arguments")
+#define CHECK_NUM_ARGS(expected) {\
+    if (getNumArguments() != expected) {\
+        SIGNAL_ERROR(ERR_NUM_ARGS);\
+        return;\
+    }\
+}
+
+ERROR_MSG(ERR_NOT_PAYABLE, "attempted to transfer funds via a non-payable function")
+
+#define CHECK_NOT_PAYABLE() {\
+    int callValue = bigIntNew(0);\
+    bigIntGetCallValue(callValue);\
+    int zero = bigIntNew(0);\
+    if (bigIntCmp(callValue, zero) > 0) {\
+        SIGNAL_ERROR(ERR_NOT_PAYABLE);\
+        return;\
+    }\
+}
```

Comment:
Consider defining `CHECK_NOT_PAYABLE` as a simple function calling the `SIGNAL_ERROR` macro. We should use macros sparingly.

Reply:
This is from https://github.com/ElrondNetwork/sc-examples/tree/master/erc20-c.
Perhaps take the task / TODO separately?